litestar/utils/sync.py

```diff
@@ -1,23 +1,44 @@
 from __future__ import annotations
 
-from collections.abc import AsyncGenerator, Awaitable, Iterable, Iterator
+from collections.abc import AsyncGenerator, Awaitable, Generator, Iterable
 from typing import (
+    TYPE_CHECKING,
     Callable,
-    Generic,
-    TypeVar,
     overload,
 )
 
-from typing_extensions import ParamSpec
+from typing_extensions import ParamSpec, TypeVar
 
 from litestar.concurrency import sync_to_thread
+from litestar.exceptions import ClientDisconnectException
 from litestar.utils.predicates import is_async_callable
 
-__all__ = ("AsyncCallable", "AsyncIteratorWrapper", "ensure_async_callable", "is_async_callable")
+if TYPE_CHECKING:
+    from types import TracebackType
+
+__all__ = ("AsyncCallable", "AsyncGeneratorWrapper", "ensure_async_callable", "is_async_callable")
 
 
 P = ParamSpec("P")
 T = TypeVar("T")
+S = TypeVar("S", default=None)
+
+
+def iterable_to_generator(iterator: Iterable[T]) -> Generator[T, S, None]:
+    """Convert an iterable to a generator.
+
+    Args:
+        iterator: An iterable.
+
+    Returns:
+        A generator.
+    """
+    for item in iterator:
+        try:
+            yield item
+
+        except ClientDisconnectException:
```

Comment:
This should not be handled at this place. It's a general utility, so it should not be concerned with streaming semantics. 

Also, I'm not sure this is required at all? If the response gets passed an async iterator, there's nothing to `throw` into to begin with, and all this wrapper does is stop iterating, which is something the calling code does anyway on disconnect